core/inx/component.go

```diff
@@ -0,0 +1,86 @@
+package inx
+
+import (
+	"context"
+
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	"go.uber.org/dig"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/gohornet/inx-indexer/pkg/daemon"
+	"github.com/gohornet/inx-indexer/pkg/nodebridge"
+	"github.com/iotaledger/hive.go/app"
+	inx "github.com/iotaledger/inx/go"
+)
+
+func init() {
+	CoreComponent = &app.CoreComponent{
+		Component: &app.Component{
+			Name:     "INX",
+			DepsFunc: func(cDeps dependencies) { deps = cDeps },
+			Params:   params,
+			Provide:  provide,
+			Run:      run,
+		},
+	}
+}
+
+type dependencies struct {
+	dig.In
+	NodeBridge *nodebridge.NodeBridge
+	Connection *grpc.ClientConn
+}
+
+var (
+	CoreComponent *app.CoreComponent
+	deps          dependencies
+)
+
+func provide(c *dig.Container) error {
+
+	type inxDepsOut struct {
+		dig.Out
+		Connection *grpc.ClientConn
+		INXClient  inx.INXClient
+	}
+
+	if err := c.Provide(func() (inxDepsOut, error) {
+		conn, err := grpc.Dial(ParamsINX.Address,
+			grpc.WithChainUnaryInterceptor(grpc_retry.UnaryClientInterceptor(), grpc_prometheus.UnaryClientInterceptor),
+			grpc.WithStreamInterceptor(grpc_prometheus.StreamClientInterceptor),
+			grpc.WithTransportCredentials(insecure.NewCredentials()),
+		)
+		if err != nil {
+			return inxDepsOut{}, err
+		}
+		client := inx.NewINXClient(conn)
```

Comment:
Update this with the pattern from `inx-mqtt` where the connection is kept internal to the nodebridge